packages/shared/src/components/search/SearchBarSuggestion.tsx

```diff
@@ -39,7 +39,7 @@ export const SearchBarSuggestion = ({
 
     // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [props.isHistory, props.suggestion?.id]);
```

Comment:
With this in place we need a local state to ensure it's only called once 🙏

Reply:
I could add a variable using `useMemo` for this, but isn't this check taking care of making sure the event is sent only once per suggestion? https://github.com/dailydotdev/apps/blob/bddb648fb3e059798b3b8e4f2750f08fb30e6096/packages/shared/src/components/search/SearchBarSuggestion.tsx#L30

Reply:
actually, nevermind, I see what you mean

Reply:
Isn't it that search suggestions don even have `id`?

Reply:
@capJavert hmm, in that case what would the `target_id` be here? https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=1166511776

Reply:
Not sure about event setup but if I am looking at `SEARCH_POST_RECOMMENDATION` query, it does not have an id.

Reply:
I thought you meant the resource actually has no ID 😅 , I added the ID to the query, thanks for pointing it out

Reply:
It's a DB ID so it will fail if we redirect with it

Reply:
But looks like we do need it for analytics confirmed by Ido here https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit?disco=AAAA47IXf14, ok, from my point of view this one is good but we do have to make sure we don't use it for recommendations. Maybe update the types to differentiate from search history and `id` there.

Reply:
Like make it `sessionId` and `id`.

Reply:
Currently `SearchSession` is used for both but `id` means different thing in regards to:
- history => session ID from previous search
- recommendation => internal API id for suggestion